google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/TraceContextInterceptor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spanner.spi.v1;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall.SimpleForwardingClientCall;
+import io.grpc.ForwardingClientCallListener.SimpleForwardingClientCallListener;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.TextMapSetter;
+
+public class TraceContextInterceptor implements ClientInterceptor  {
+
+    private final ContextPropagators propagators;
+
+    TraceContextInterceptor(OpenTelemetry openTelemetry) {
+        this.propagators = openTelemetry.getPropagators();
```

Comment:
This approach means that users are required to create and register the correct propagators with `OpenTelemetry` for this to work. That has both an advantage and disadvantage:
1. It won't work automatically out of the box without the user adding some additional (simple) configuration.
2. It gives users a simple opt-out of the feature.

Is the above intentional? Or do we want this to always be enabled without any additional configuration from the user when OpenTelemetry is being used?

Reply:
Yes, this is intentional. In the documentation, it will be mentioned that user need to set the W3TraceContextPropagator(this provide methods to inject the `traceparent` header from SpanContext) for server side tracing to work.